### DISCUSSION TITLE
Concept about FWUPD_DEVICE_FLAG_DUAL_IMAGE flag?

### DISCUSSION THREAD

**Discussion by lingc24** [fwupd/fwupd#6778] [General]
Hi,
I'm wondering if the following situation fit the concept of FWUPD_DEVICE_FLAG_DUAL_IMAGE flag?

There are 2 bank in the flash (bank0 and bank1). The bank0 space could be updated to the latest firmware version each time. The bank1 contains the default firmware version. If the firmware update is failed, the device would start by bank1 and the firmware version is the default one.

Thanks

**Comment by hughsie** [comment]
Yes, that's fine -- what dual bank is conceptually saying is "if I corrupt the firmware update, is the device going to behave normally, or is it going to be stuck in bootloader/recovery mode".

**Reply by lingc24** [reply]
Thanks for the explanation.
I'll add this flag at next pull request.
